### PR TITLE
Verify stack and module existence on some creations

### PR DESCRIPTION
**Pull request by marcinwyszynski**
## Description of the change

When a resource can be attached to either a stack or a module, validate the existence to prevent creating a resource for the wrong entity.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [x] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer
- [ ] The target branch is `future` unless the change is going directly into production

### CODE REVIEW

**Comment by marcinwyszynski** [comment]
Task linked: [CU-1yyac9w In the provider, adding things to a stack instead of a module causes a weird result](https://app.clickup.com/t/1yyac9w)

File: spacelift/verify_project.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/shurcooL/graphql"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
```
Comment (b4k3r) [inline spacelift/verify_project.go:8]: nit: imports

**Review by b4k3r — APPROVED** [review]

File: spacelift/verify_project.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+func verifyStack(ctx context.Context, stackID string, meta interface{}) error {
+	var query struct {
+		Stack *structs.Stack `graphql:"stack(id: $id)"`
+	}
+
+	variables := map[string]interface{}{"id": graphql.ID(stackID)}
+
+	if err := meta.(*internal.Client).Query(ctx, "ModuleVerifyExistence", &query, variables); err != nil {
+		return errors.Wrap(err, "could not query for module")
```
Comment (b4k3r) [inline spacelift/verify_project.go:38]: could not query for stack

File: spacelift/verify_project.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+func verifyStack(ctx context.Context, stackID string, meta interface{}) error {
+	var query struct {
+		Stack *structs.Stack `graphql:"stack(id: $id)"`
+	}
+
+	variables := map[string]interface{}{"id": graphql.ID(stackID)}
+
+	if err := meta.(*internal.Client).Query(ctx, "ModuleVerifyExistence", &query, variables); err != nil {
```
Comment (b4k3r) [inline spacelift/verify_project.go:37]: StackVerifyExistence

File: spacelift/verify_project.go (hunk 1 above)
Comment (marcinwyszynski) [reply spacelift/verify_project.go:9]: Thanks, fixed.

File: spacelift/verify_project.go (hunk 3 above)
Comment (marcinwyszynski) [reply spacelift/verify_project.go]: Thanks, fixed.

File: spacelift/verify_project.go (hunk 2 above)
Comment (marcinwyszynski) [reply spacelift/verify_project.go]: Thanks, fixed.

**Review by adamconnelly — APPROVED** [review]

**Pull request merged**